trin-core/src/portalnet/storage.rs

```diff
@@ -24,82 +26,217 @@ pub enum DistanceFunction {
+        if !self.is_key_within_radius(&key) {
+            let distance = self.distance_to_key(&key);
+            return Err(PortalContentStoreError::InsufficientRadius {
+                radius: self.radius,
+                distance,
+            });
+        }
+
+        let content_id = key.content_id();
+        let value: &[u8] = value.as_ref();
+        self.store.insert(content_id.to_vec(), value.to_vec());
+
+        Ok(())
+    }
+
+    fn is_key_within_radius<K: OverlayContentKey>(&self, key: &K) -> bool {
+        let distance = self.distance_to_key(key);
+        distance <= self.radius
+    }
+
+    fn radius(&self) -> U256 {
+        self.radius
+    }
+}
+
+/// Struct for configuring a `PortalStorage` instance.
 #[derive(Clone)]
 pub struct PortalStorageConfig {
     pub storage_capacity_kb: u64,
     pub node_id: NodeId,
-    pub distance_function: DistanceFunction,
+    pub distance_fn: DistanceFunction,
     pub db: Arc<rocksdb::DB>,
     pub sql_connection_pool: Pool<SqliteConnectionManager>,
 }
 
-/// Struct whose public methods abstract away Kademlia-based storage behavior.
+/// Struct whose public methods abstract away Kademlia-based store behavior.
 pub struct PortalStorage {
     node_id: NodeId,
     storage_capacity_in_bytes: u64,
-    // pub to allow for tests in trin-history/src/content_key.rs
-    pub data_radius: u64,
+    radius: U256,
     farthest_content_id: Option<[u8; 32]>,
     db: Arc<rocksdb::DB>,
     sql_connection_pool: Pool<SqliteConnectionManager>,
-    distance_function: DistanceFunction,
+    distance_fn: DistanceFunction,
 }
 
-/// Error type returned in a Result by any failable public PortalStorage methods.
-#[derive(Debug, Error)]
-pub enum PortalStorageError {
-    #[error("RocksDB Error")]
-    RocksDB(#[from] rocksdb::Error),
-
-    #[error("Sqlite Error")]
-    Sqlite(#[from] rusqlite::Error),
-
-    #[error("Sqlite Connection Pool Error")]
-    SqliteConnectionPool(#[from] r2d2::Error),
-
-    #[error("IO Error")]
-    IOError(#[from] std::io::Error),
-
-    #[error("Sum data size error: received None from SQLite")]
-    SumError(),
-
-    #[error("While {doing:?}, expected to receive data of size {expected:?} but found data of size {actual:?}")]
-    DataSizeError {
-        doing: String,
-        expected: usize,
-        actual: usize,
-    },
-
-    #[error("String error value returned from {function_name:?}: {error:?}")]
-    StringError {
-        function_name: String,
-        error: std::ffi::OsString,
-    },
+impl PortalContentStore for PortalStorage {
+    fn get<K: OverlayContentKey>(
+        &self,
+        key: &K,
+    ) -> Result<Option<Vec<u8>>, PortalContentStoreError> {
+        let key = key.content_id();
+        let value = self.db.get(key)?;
+        Ok(value)
+    }
 
-    #[error("Content can not be stored since it falls outside our radius")]
-    OutsideDistanceError,
+    fn put<K: OverlayContentKey, V: AsRef<[u8]>>(
```

Comment:
This function feels somewhat redundant. Why aren't we just using `store()`?

Reply:
Well whether we call it `store` or `put`, we need to implement the trait. I'm not sure if you mean that we should just call this method `store` in the trait definition. For a comment on that, see comment thread above.

Reply:
For sure, I'd lean towards using the same name for both the function in the trait, and the one it proxies towards, whatever naming we end up with